### PR TITLE
checked_add using more native overflow detection.

### PR DESCRIPTION
**Pull request by devnexen**

### CODE REVIEW

File: src/lib/utils/safeint.h [hunk 1]
```diff
@@ -25,9 +25,14 @@ class Integer_Overflow_Detected final : public Exception
 
 inline size_t checked_add(size_t x, size_t y, const char* file, int line)
    {
-   // TODO: use __builtin_x_overflow on GCC and Clang
+#if defined(BOTAN_BUILD_COMPILER_IS_GCC) || defined(BOTAN_BUILD_COMPILER_IS_CLANG)
```
Comment (randombit) [inline src/lib/utils/safeint.h:28]: Use `BOTAN_COMPILER_HAS_BUILTIN`

**Comment by randombit** [comment]
Some build problems on 32 bit systems. I think `__builtin_add_overflow` will dispatch to the correct version.

**Comment by devnexen** [comment]
yes

**Review by randombit — APPROVED** [review]
Nice

**Pull request merged**